core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java

```diff
@@ -378,6 +382,12 @@ public void start()
                     // query already started or finished
                     return;
                 }
+                Thread planningThread = currentThread();
+                stateMachine.getStateChange(PLANNING).addListener(() -> {
+                    if (stateMachine.getQueryState() == FAILED) {
```

Comment:
Does cancellation always transition to failed state?

Reply:
Both `transitionToCancelled` and `transitionToFailed` set state to `FAILED`